### PR TITLE
Migrate concourse variables

### PR DESCRIPTION
**Pull request by yimngli-vmware**

### CODE REVIEW

File: concourse/pipelines/templates/get_pivnet_artifacts-tpl.yml [hunk 1]
```diff
@@ -19,7 +19,7 @@ resources:
   icon: git
   source:
     branch: ((pxf-git-branch))
-    uri: ((pxf-git-remote))
+    uri: ((ud/pxf/prod/git-remote))
```
Comment (denalex) [inline concourse/pipelines/templates/get_pivnet_artifacts-tpl.yml:22]: are they really different between dev and prod ?

File: concourse/pipelines/templates/perf_pipeline-tpl.yml [hunk 2]
```diff
@@ -330,36 +330,36 @@ jobs:
...
+      ADL_ACCOUNT: ((adl-account))
+      ADL_CLIENT_ID: ((adl-client-id))
+      ADL_CREDENTIAL: ((adl-credential))
+      ADL_REFRESH_URL: ((adl-refresh-url))
+      AWS_ACCESS_KEY_ID: ((tf-machine-access-key-id))
+      AWS_SECRET_ACCESS_KEY: ((tf-machine-secret-access-key))
+      AWS_DEFAULT_REGION: ((ud/common/aws-region))
+      GOOGLE_CREDENTIALS: ((ud/pxf/secrets/data-gpdb-ud-google-json-key))
+      WASB_ACCOUNT_NAME: ((wasb-account-name))
+      WASB_ACCOUNT_KEY: ((wasb-account-key))
```
Comment (denalex) [inline concourse/pipelines/templates/perf_pipeline-tpl.yml:363]: should these also come from `ud/pxf/common/` and `ud/pxf/secrets/` respectively ?

File: concourse/pipelines/templates/pr_pipeline-tpl.yml [hunk 3]
```diff
@@ -275,8 +275,8 @@ jobs:
     image: gpdb[[gp_ver]]-pxf-dev-ubuntu18-image
     file: pxf_src/concourse/tasks/build.yml
     params:
-      LICENSE: ((pxf-rpm-license))
-      VENDOR: ((pxf-rpm-vendor))
+      LICENSE: ((ud/pxf/common/rpm-license))
+      VENDOR: ((ud/pxf/common/rpm-vendor))
```
Comment (denalex) [inline concourse/pipelines/templates/pr_pipeline-tpl.yml:279]: did you use `deb-maintainer` in another file instead ?

File: automation/jsystem.properties [hunk 4]
```diff
@@ -13,5 +13,7 @@ reporter.classes=jsystem.extensions.report.html.LevelHtmlTestReporter;jsystem.fr
 resources.src=/home/gpadmin/workspace/pxf/automation/src/main/resources
 sutClassName=jsystem.framework.sut.SutImpl
 sutFile=default.xml
-tests.dir=/home/gpadmin/workspace/pxf/automation/target/test-classes
+
+tests.dir=/Users/yimingli/workspace/pxf/automation/target/test-classes
```
Comment (denalex) [inline automation/jsystem.properties:17]: this line will be updated when you run the automation tests on your workstation, we usually avoid merging this change in, please revert to the original

File: concourse/pipelines/templates/build_pipeline-tpl.yml [hunk 5]
```diff
@@ -272,101 +272,101 @@ resources:
...
     json_key: ((concourse-gcs-resources-service-account-key))
-    regexp: ((pxf-releng-drop-path))/gpdb6/pxf-gp6-(.*)-el7.x86_64.tar.gz
+    regexp: ((ud/pxf/common/releng-drop-path))/gpdb6/pxf-gp6-(.*)-el7.x86_64.tar.gz
 
 ## ---------- PXF ShipIt resources ----------
 - name: pxf_shipit_file
   type: gcs
   icon: google-drive
   source:
     bucket: data-gpdb-ud-pxf-releases
```
[the remark this answers is not in the document]
Comment (yimngli-vmware) [reply concourse/pipelines/templates/build_pipeline-tpl.yml:356]: create a secret for bucket

File: concourse/pipelines/templates/build_pipeline-tpl.yml [hunk 6]
```diff
@@ -533,8 +533,9 @@ resources:
   icon: google-drive
   source:
     bucket: data-gpdb-ud-pxf-build
-    json_key: ((pxf-storage-service-account-key))
-    versioned_file: ((pxf-build-bucket-prefix))/snapshots/pxf6/pxf-gp[[gp_ver]].el7.tar.gz
+    json_key: ((ud/pxf/secrets/pxf-storage-service-account-key))
+    versioned_file: ((ud/pxf/[[ pipeline_type ]]/build-bucket-prefix))/snapshots/pxf6/pxf-gp[[gp_ver]].el7.tar.gz
+    versioned_file: [[ pipeline_type ]]/snapshots/pxf6/pxf-gp[[gp_ver]].el7.tar.gz
```
[the remark this answers is not in the document]
Comment (yimngli-vmware) [reply concourse/pipelines/templates/build_pipeline-tpl.yml:538]: don't need to store versioned_file in vault

File: concourse/pipelines/templates/build_pipeline-tpl.yml [hunk 7]
```diff
@@ -562,26 +563,26 @@ resources:
...
+      region_name: ((ud/common/aws-region))
+      bucket: ((ud/pxf/common/tf-bucket-name))
       bucket_path: ((tf-bucket-path))
 
 - name: terraform_ipa_hadoop
   type: terraform-0.14.10
   source:
     env:
-      GOOGLE_CREDENTIALS: ((data-gpdb-ud-ipa-google-service-account-key))
+      GOOGLE_CREDENTIALS: ((ud/pxf/secrets/data-gpdb-ud-ipa-google-service-account-key))
```
[the remark this answers is not in the document]
Comment (yimngli-vmware) [reply concourse/pipelines/templates/build_pipeline-tpl.yml:580]: data-gpdb-ud -> google-project-id

File: concourse/pipelines/templates/build_pipeline-tpl.yml [hunk 8]
```diff
@@ -1733,11 +1734,11 @@ jobs:
     image: google-cloud-sdk-slim-image
     file: pxf_src/concourse/tasks/get_pxf_release_artifacts.yml
     params:
-      GCS_ODP_PATH: ((pxf-releases-bucket-prefix))/ODP
-      GCS_OSL_PATH: ((pxf-releases-bucket-prefix))/OSL
+      GCS_ODP_PATH: [[ pipeline_type ]]/ODP
+      GCS_OSL_PATH: [[ pipeline_type ]]/OSL
       GCS_RELEASES_BUCKET: data-gpdb-ud-pxf-releases
```
[the remark this answers is not in the document]
Comment (yimngli-vmware) [reply concourse/pipelines/templates/build_pipeline-tpl.yml:1739]: replace data-gpdb-ud

File: concourse/pipelines/templates/pr_pipeline-tpl.yml (hunk 3 above)
Comment (yimngli-vmware) [reply concourse/pipelines/templates/pr_pipeline-tpl.yml:279]: change rpm-vendor into deb-maintainer for all the ubuntu tasks

File: concourse/pipelines/singlecluster-pipeline.yml [hunk 9]
```diff
@@ -15,72 +15,72 @@ resources:
...
       versioned_file: jdbc/postgresql-jdbc-8.4.704.jar
 
   - name: singlecluster-HDP2-gcs
     type: google-cloud-storage
     icon: google-drive
     source:
-      bucket: {{pxf-build-resources-bucket-name}}
-      json_key: {{data-gpdb-ud-google-json-key}}
+      bucket: ((ud/pxf/common/build-resources-bucket-name))
+      json_key: ((ud/pxf/secrets/data-gpdb-ud-google-json-key))
```
[the remark this answers is not in the document]
Comment (yimngli-vmware) [reply concourse/pipelines/singlecluster-pipeline.yml:59]: find out the project name for this json key and replace the name

File: concourse/pipelines/templates/build_pipeline-tpl.yml (hunk 7 above)
[the remark this answers is not in the document]
Comment (yimngli-vmware) [reply concourse/pipelines/templates/build_pipeline-tpl.yml:580]: ipa-ccp-ci-service

File: concourse/pipelines/templates/build_pipeline-tpl.yml [hunk 10]
```diff
@@ -134,9 +134,9 @@ anchors:
     platform: linux
   image: ccp-7-image
   params:
-    GOOGLE_CREDENTIALS: ((data-gpdb-ud-kerberos-google-service-account-key))
-    GOOGLE_PROJECT_ID: ((data-gpdb-ud-kerberos-google-project-id))
-    GOOGLE_ZONE: ((data-gpdb-ud-kerberos-google-zone))
+    GOOGLE_CREDENTIALS: ((ud/pxf/secrets/data-gpdb-ud-kerberos-google-service-account-key))
+    GOOGLE_PROJECT_ID: ((ud/pxf/common/data-gpdb-ud-kerberos-google-project-id))
+    GOOGLE_ZONE: ((ud/pxf/common/data-gpdb-ud-kerberos-google-zone))
```
[the remark this answers is not in the document]
Comment (yimngli-vmware) [reply concourse/pipelines/templates/build_pipeline-tpl.yml:139]: ccp-ci-service  kerberos-ccp-ci-service

File: concourse/pipelines/singlecluster-pipeline.yml (hunk 9 above)
[the remark this answers is not in the document]
Comment (yimngli-vmware) [reply concourse/pipelines/singlecluster-pipeline.yml:59]: data-gpdb-ud-google-json-key -> google-project-id-ccp-ci-service-account-key

File: concourse/Makefile [hunk 11]
```diff
@@ -286,8 +252,25 @@ query-execution-parquet-1g:
...
-		TEAM_SECRETS="--load-vars-from=${HOME}/workspace/gp-continuous-integration/secrets/pxf-perf-secrets.yml"
+		INPUT_FILE=ccp-provision-from-release-pxf.yml
+
+# ============================= LONGEVITY PIPELINE TARGETS =============================
+.PHONY: set-longevity-pipeline
+set-longevity-pipeline:
+	$(FLY_CMD) --target=$(CONCOURSE) \
+		set-pipeline \
+		--check-creds \
+		--pipeline=dev:longevity_$(YOUR_TAG)_6X_STABLE \
```
Comment (denalex) [inline concourse/Makefile:263]: I think we must fail if `YOUR_TAG` is not defined ?

File: concourse/pipelines/certification_pipeline.yml [hunk 12]
```diff
@@ -40,16 +40,16 @@ resources:
   type: gcs
   icon: google-drive
   source:
-    bucket: data-gpdb-ud-pxf-build-resources
-    json_key: {{pxf-storage-service-account-key}}
+    bucket: ((ud/pxf/common/build-resources-bucket-name))
```
Comment (denalex) [inline concourse/pipelines/certification_pipeline.yml:43]: I believe in other projects we place non-secrets into GIT under  `ci/vars/common.yml`. Should we adopt the same approach for PXF ?

File: concourse/pipelines/longevity_pipeline.yml [hunk 13]
```diff
@@ -45,25 +45,25 @@ resources:
   icon: tractor
   source:
     env:
-      AWS_ACCESS_KEY_ID: {{tf-machine-access-key-id}}
-      AWS_SECRET_ACCESS_KEY: {{tf-machine-secret-access-key}}
-      GOOGLE_CREDENTIALS: {{google-service-account-key}}
+      AWS_ACCESS_KEY_ID: ((tf-machine-access-key-id))
+      AWS_SECRET_ACCESS_KEY: ((tf-machine-secret-access-key))
+      GOOGLE_CREDENTIALS: ((ud/pxf/secrets/google-service-account-key))
```
Comment (denalex) [inline concourse/pipelines/longevity_pipeline.yml:50]: should this be `pxf-storage-service-account-key` ?

File: concourse/pipelines/pg_regress_pipeline.yml [hunk 14]
```diff
@@ -254,7 +254,7 @@ jobs:
       - name: pxf_tarball
       params:
         ACCESS_KEY_ID: {{tf-machine-access-key-id}}
-        GOOGLE_CREDENTIALS: {{data-gpdb-ud-google-json-key}}
+        GOOGLE_CREDENTIALS: ((ud/pxf/secrets/google-project-id-ccp-ci-service-account-key))
```
Comment (denalex) [inline concourse/pipelines/pg_regress_pipeline.yml:257]: Didn't we decide to just use `ud/pxf/secrets/ccp-ci-service-account-key` for the default google project and `ud/pxf/secrets/ipa-ccp-ci-service-account-key` and `ud/pxf/secrets/kerberos-ccp-ci-service-account-key` respectively for others ?

File: concourse/pipelines/templates/build_pipeline-tpl.yml [hunk 15]
```diff
@@ -1803,21 +1803,22 @@ jobs:
...
+        file: pxf_artifacts/((ud/pxf/common/osl-file-prefix))*.txt
 {% if email_notification %}
   - put: send-an-email
     params:
       subject: pxf_artifacts/email_subject.txt
       body: pxf_artifacts/email_body.txt
-      attachment_globs: [pxf_artifacts/((pxf-osl-file-prefix))*.txt]
-      to_text: "((pxf-releng-email)),((pxf-ud-email))"
+      attachment_globs: [pxf_artifacts/((ud/pxf/common/osl-file-prefix))*.txt]
+      # TODO: how does this work on master w/ dev?
```
Comment (denalex) [inline concourse/pipelines/templates/build_pipeline-tpl.yml:1822]: do we still need TODO ?

File: concourse/pipelines/templates/perf_pipeline-tpl.yml [hunk 16]
```diff
@@ -39,15 +39,15 @@ set_failed_gpdb_anchor: &set_failed
...
-          GOOGLE_CREDENTIALS: {{google-service-account-key}}
-          GOOGLE_PROJECT_ID: {{google-project-id}}
-          GOOGLE_ZONE: {{google-zone}}
-          GOOGLE_SERVICE_ACCOUNT: {{google-service-account}}
-          AWS_ACCESS_KEY_ID: {{tf-machine-access-key-id}}
-          AWS_SECRET_ACCESS_KEY: {{tf-machine-secret-access-key}}
-          AWS_DEFAULT_REGION: {{tf-machine-region}}
-          BUCKET_PATH: {{tf-bucket-path}}
-          BUCKET_NAME: {{tf-bucket-name}}
+          GOOGLE_CREDENTIALS: ((ud/pxf/secrets/google-service-account-key))
```
Comment (denalex) [inline concourse/pipelines/templates/perf_pipeline-tpl.yml:42]: should this be `pxf-storage-...`

File: concourse/Makefile (hunk 11 above)
Comment (ashuka24) [reply concourse/Makefile:263]: Could we set a default value at the top of this Makefile, similar to how we do the others?

File: concourse/pipelines/templates/dev_build_pipeline-tpl.yml [hunk 17]
```diff
@@ -566,14 +566,14 @@ jobs:
   plan:
   - in_parallel:
     - get: pxf_src
-      passed: [Testing Gate for PXF-GP]
```
Comment (ashuka24) [inline concourse/pipelines/templates/dev_build_pipeline-tpl.yml:569]: why was the passing condition changed here?

Comment (bradfordb-vmware) [reply concourse/pipelines/templates/dev_build_pipeline-tpl.yml:569]: The dev pipeline doesn't contain a testing gate -- I guess no one has flown a dev pipeline with HDP2 and noticed this issue before.

File: concourse/pipelines/certification_pipeline.yml [hunk 18]
```diff
@@ -212,15 +212,15 @@ jobs:
     file: pxf_src/concourse/tasks/test_certification.yml
     image: gpdb5-pxf-dev-centos7-image
     params:
-      ACCESS_KEY_ID: {{tf-machine-access-key-id}}
+      ACCESS_KEY_ID: ((tf-machine-access-key-id))
       GP_VER: 5
       GROUP: gpdb,proxy,profile
-      SECRET_ACCESS_KEY: {{tf-machine-secret-access-key}}
+      SECRET_ACCESS_KEY: ((tf-machine-secret-access-key))
```
Comment (ashuka24) [inline concourse/pipelines/certification_pipeline.yml:218]: should this key be in vault?

Comment (bradfordb-vmware) [reply concourse/pipelines/certification_pipeline.yml:218]: It is, but its a GPDB shared secret no `ud/pxf/*` prefix.

File: concourse/pipelines/singlecluster-pipeline.yml [hunk 19]
```diff
@@ -15,72 +15,72 @@ resources:
...
       versioned_file: jdbc/postgresql-jdbc-8.4.704.jar
 
   - name: singlecluster-HDP2-gcs
     type: google-cloud-storage
     icon: google-drive
     source:
-      bucket: {{pxf-build-resources-bucket-name}}
-      json_key: {{data-gpdb-ud-google-json-key}}
+      bucket: ((ud/pxf/common/build-resources-bucket-name))
+      json_key: ((ud/pxf/secrets/gsc-ci-service-account-key))
```
Comment (denalex) [inline concourse/pipelines/singlecluster-pipeline.yml:59]: what is `gsc-ci-service-account-key` ? is this for Spark project ? should we be using pif-specific service accounts and its keys here ? Will `pxf-storage-service-account-key` be sufficient here ?

File: concourse/pipelines/templates/build_pipeline-tpl.yml [hunk 20]
```diff
@@ -272,101 +272,101 @@ resources:
...
+    branch: ((ud/pxf/common/gp-concourse-cluster-provisioner-git-branch))
+    private_key: ((gp-concourse-cluster-provisioner-git-key))
+    uri: ((ud/pxf/common/gp-concourse-cluster-provisioner-git-remote))
 
 ## ---------- RelEng Drop Artifacts ----------
 - name: pxf_gp5_rhel7_release_to_releng
   type: gcs
   source:
-    bucket: ((pxf-releng-drop-bucket))
+    bucket: ((ud/pxf/[[ environment ]]/releng-drop-bucket))
```
Comment (denalex) [inline concourse/pipelines/templates/build_pipeline-tpl.yml:283]: I see in another place above:
`bucket: ((ud/pxf/common/build-resources-bucket-name))`

So should we stick to convention of using the suffix `-bucket-name` for names of buckets ?

File: concourse/pipelines/templates/build_pipeline-tpl.yml [hunk 21]
```diff
@@ -1158,7 +1158,7 @@ jobs:
     file: pxf_src/concourse/tasks/test.yml
     image: gpdb[[gp_ver]]-pxf-dev-centos7-image
     params:
-      GOOGLE_CREDENTIALS: ((data-gpdb-ud-google-json-key))
+      GOOGLE_CREDENTIALS: ((ud/pxf/secrets/gsc-ci-service-account-key))
```
Comment (denalex) [inline concourse/pipelines/templates/build_pipeline-tpl.yml:1161]: same

File: concourse/pipelines/templates/external-table-tpl.yml [hunk 22]
```diff
@@ -31,9 +31,9 @@ resources:
 {% endif %}
     tag: latest
     username: _json_key
-    password: ((data-gpdb-ud-google-json-key))
+    password: ((ud/pxf/secrets/gsc-ci-service-account-key))
```
Comment (denalex) [inline concourse/pipelines/templates/external-table-tpl.yml:34]: same

File: concourse/pipelines/singlecluster-pipeline.yml (hunk 19 above)
Comment (yimngli-vmware) [reply concourse/pipelines/singlecluster-pipeline.yml:59]: according to `gpdb_common-ci-secrets.yml` and `pxf-secrets.yml`, `pxf-storage-service-account-key`, `data-gpdb-ud-google-json-key` and `google-service-account-key` have different values, and  `data-gpdb-ud-google-json-key` is "Google Cloud Access Key for pxf/UD". 

The client email for `pxf-storage-service-account-key` is `pxf-storage-service-account@data-gpdb-ud.iam.gserviceaccount.com`

email for `google-service-account-key` is `ccp-ci-service@data-gpdb-ud.iam.gserviceaccount.com`

and for `data-gpdb-ud-google-json-key` the email is `gsc-ci-service@data-gpdb-ud.iam.gserviceaccount.com`, so I just take the email name.

File: concourse/pipelines/certification_pipeline.yml (hunk 12 above)
Comment (yimngli-vmware) [reply concourse/pipelines/certification_pipeline.yml:43]: Will do the same in the future

File: concourse/pipelines/templates/perf_pipeline-tpl.yml (hunk 16 above)
Comment (yimngli-vmware) [reply concourse/pipelines/templates/perf_pipeline-tpl.yml]: it's outdated now but it should be `ccp-ci-service-account-key`

File: concourse/pipelines/longevity_pipeline.yml (hunk 13 above)
Comment (yimngli-vmware) [reply concourse/pipelines/longevity_pipeline.yml]: shoule be `ccp-ci-service-account-key`

**Pull request merged**